### PR TITLE
fix(completions): Subcommand completions are now wrapped in quotes for `which`

### PR DESCRIPTION
**Pull request by Mrfiregem**
## Description

When tab-completing commands with spaces in them as parameters to `which`, completed arguments lacked quotes, treating the text as two (or more) commands instead.

This PR aims to fix that by quoting completed `which` arguments when needed. It also modifies the completion for `@complete` to do the same thing.

### Subcommand completions are now wrapped in quotes for `which`

Tab completions for `which` and `attr complete`/`@complete` commands will now properly quote command names containing spaces.

<table>
<tr>
<th>Before</th>
<th>After</th>
</tr>

<tr>
<td>

```nushell
> which 'path par  # Tab pressed
> which path parse
╭───┬─────────┬──────┬──────────╮
│ # │ command │ path │   type   │
├───┼─────────┼──────┼──────────┤
│ 0 │ path    │      │ built-in │
│ 1 │ parse   │      │ built-in │
╰───┴─────────┴──────┴──────────╯
```

</td>
<td>

```nushell
> which 'path par   # Tab pressed
> which 'path parse'
╭───┬────────────┬──────┬──────────╮
│ # │  command   │ path │   type   │
├───┼────────────┼──────┼──────────┤
│ 0 │ path parse │      │ built-in │
╰───┴────────────┴──────┴──────────╯
 
```

</td>
</tr>
</table>

## Additional notes

First time working on a non-command PR. Let me know if I missed anything obvious.

### CODE REVIEW

File: crates/nu-cli/tests/completions/mod.rs [hunk 1]
```diff
@@ -770,6 +770,11 @@ fn which_command_completions() {
     #[cfg(not(windows))]
     let expected: Vec<_> = vec!["sleep", "%sleep", "^sleep"];
     match_suggestions(&expected, &suggestions);
+    // commands with spaces
+    let completion_str = "which detect";
```
Comment (blindFS) [inline crates/nu-cli/tests/completions/mod.rs]: Nitpick: I think it's better to cover edge cases like `which 'detect`/`which "detect` or even quotes in the command name itself.

I believe you've handled them quite well in the code, just feel safer to cover them in tests.

**Comment by blindFS** [comment]
LGTM overall, thanks!

File: crates/nu-cli/tests/completions/mod.rs (hunk 1 above)
Comment (Mrfiregem) [reply crates/nu-cli/tests/completions/mod.rs]: Thanks, I tested commands with quotes locally and they worked, but I just added tests for those cases as well. Let's hope they work 👍

File: crates/nu-cli/src/completions/command_completions.rs [hunk 2]
```diff
@@ -9,14 +9,28 @@ use reedline::Suggestion;
...
+        return name;
+    }
+    if name.contains('\'') {
+        format!("\"{}\"", name.replace('\\', "\\\\").replace('"', r#"\""#))
+    } else if name.contains(' ') || name.contains('"') {
+        format!("'{name}'")
+    } else {
+        name
+    }
+}
```
Comment (Bahex) [inline crates/nu-cli/src/completions/command_completions.rs]: You can use functions from `nu_utils::quoting` instead, which cover edge cases you might have missed and have tests.

Also, there is a similar function here https://github.com/Mrfiregem/nushell/blob/cd7a3d5e82dace45a2bdc19218fb79877875791e/crates/nu-cli/src/completions/exportable_completions.rs#L16-L26

Can you replace that one with `nu_utils::quoting` too while you're at it?

**Review by Bahex — CHANGES_REQUESTED** [review]

File: crates/nu-cli/src/completions/command_completions.rs (hunk 2 above)
Comment (Mrfiregem) [reply crates/nu-cli/src/completions/command_completions.rs]: Thanks for letting me know about this! I've converted both functions to use that module's functions.

**Review by Bahex — APPROVED** [review]
